app/models/experimentation/experiment.rb

```diff
@@ -12,6 +12,7 @@ class Experiment < ApplicationRecord
     validates :experiment_type, presence: true
     validate :validate_date_range
     validate :one_active_experiment_per_type
+    validate :validate_filters
```

Comment:
I'll add unit tests for this before merging. Have tested a bit manually though.

Reply:
Can you please add the unit tests as well? It would help to understand the logic better

Reply:
Just pushed unit tests for validation filter. Will push tests for eligible_patients in a while.